File: poc-gradle/src/main/java/br/com/bra/pocgradle/kafka/settings/LocalKafkaProperties.java
package br.com.bra.pocgradle.kafka.settings;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

import java.util.List;

@Data
@Configuration
@ConfigurationProperties(prefix = "kafka-config")
public class LocalKafkaProperties {
    private String serverUrl;
    private String schemaRegistryUrl;
    private String groupId;
    private Integer concurrency;
    private List<KafkaTopicModel> topics;
}

File: processing-service/src/main/kotlin/br/com/bra/processingservice/kafka/settings/LocalKafkaProperties.kt
package br.com.bra.processingservice.kafka.settings

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "kafka-config")
data class LocalKafkaProperties(
    var serverUrl: String = "",
    var schemaRegistryUrl: String = "",
    var groupId: String = "",
    var concurrency: Int = 0
)